### PR TITLE
Resolved warning of non-finite test scores and input_shape usage in dense layer

### LINKED ISSUE #201: [BUG] GridSearchCV Warning: Non-Finite Test Scores for SVR Model
### Has this bug been raised before?

- [X] I have checked "open" AND "closed" issues and this is not a duplicate

### Description

I am encountering a warning during the hyperparameter tuning of a Support Vector Regression (SVR) model using GridSearchCV in scikit-learn. The warning states: "One or more of the test scores are non-finite: [nan nan nan nan nan nan]."
Another warning occured when attempting to define a Keras Sequential model. The warning indicates that the model is improperly configured by passing the input_shape directly to the Dense layers rather than using an Input layer at the beginning of the model.

Screenshot 2024-10-28 230852
Screenshot 2024-10-28 231629
Screenshot 2024-10-28 231822

### Steps to Reproduce

Include any relevant details like:

- What page you were on...        stock_market(complete).ipynb
- What you were trying to do...         I was executing the code when I encountered the warning
- What went wrong...       It displayed the above mentioned warnings

### Screenshots

_No response_

### Do you want to work on this issue?

Yes

### If "yes" to the above, please explain how you would technically implement this.

I will switch the scoring metric to a regression-appropriate metric and for the second warning I will use Input layer

### PR DESCRIPTION
**Pull request by shristirwt**
Issue Reference: Fixes #201 
Description of changes: I updated the code to resolve the warnings of non-Finite Values in GridSearchCV Test Scores for Support Vector Regressor (SVR) Model and input Layer Warning in Keras Sequential Model Due to input_shape Usage in Dense Layer

### CODE REVIEW

**Review by Mayureshd-18 — APPROVED** [review]
Looks good to me!

**Review by jvedsaqib — APPROVED** [review]
